vendor/engines/secure_rooms/app/controllers/secure_rooms/card_readers_controller.rb

```diff
@@ -0,0 +1,81 @@
+module SecureRooms
+
+  class CardReadersController < ApplicationController
+
+    admin_tab :all
+
+    layout "two_column"
+
+    before_action :authenticate_user!
+    before_action :check_acting_as
+    before_action :authorize_card_reader
```

Comment:
I think we're getting pretty close. I think we can use CanCan's `authorize_resource` method after we load it in `load_card_reader`, or possibly, even getting rid of `load_card_reader` and using `load_and_authorize_resource through: :product`. Then we don't need to add our own explicit authorization method.